framework/reflect.go

```diff
@@ -85,13 +102,26 @@ func (m *Import) reflectMap(mv reflect.Value) (reflect.Value, error) {
 			v = reflect.Zero(mv.Type().Elem())
 		}
 
-		mv.SetMapIndex(k, v)
+		result.SetMapIndex(k, v)
 	}
 
-	return mv, nil
+	return result, nil
 }
 
 func (m *Import) reflectSlice(v reflect.Value) (reflect.Value, error) {
+	// Create a new map for this. This avoids conflicts and panics on shared
+	// data, and ensures that we aren't altering data in the original namespace.
+	// []interface{} is always used, regardless of the actual type of the map
+	// sent along. This prevents type panics.
+	//
+	// Do a quick check to see if we have a zero-value map (nil map). If we do,
+	// return that.
+	if v == reflect.Zero(v.Type()) {
```

Comment:
As above in map, I think we are doubling up on creating a zero value. Can re-use the one from the condition.